pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -176,6 +222,36 @@ func (cloud *CloudProvider) WaitForAttach(ctx context.Context, zone, diskName, i
 	})
 }
 
+// WaitANdGetSnapshot will keep tring to get the VolumeSnapshot response until it is created or failed.
+func (cloud *CloudProvider) WaitAndGetSnapshot(ctx context.Context, snapshotName string) (*compute.Snapshot, error) {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+	timer := time.NewTimer(1*time.Minute)
+	defer timer.Stop()
+
+	for {
```

Comment:
is it possible to instead use the apimachinery `wait.Poll` here?

Reply:
I need to return snapshot object, but wait.Poll only allows to return a bool value?